gradleUtils/PatchedArtifactsSettings.gradle

```diff
@@ -0,0 +1,148 @@
+import groovy.json.JsonSlurper
+import java.util.concurrent.TimeUnit
+
+class PatchedArtifactsConfig {
+    String version = null
+    String packageName = null
+    String githubUsername = null
+    String githubToken = null
+    Boolean buildFromSource = true
+}
+
+def patchedArtifacts = settings.getExtensions().create('patchedArtifacts', PatchedArtifactsConfig)
+
+def lifecycle(message) {
+    logger.lifecycle("\u001B[34m[PatchedArtifacts] $message\u001B[0m")
+}
+
+def warn(message) {
+    logger.warn("\u001B[33m[PatchedArtifacts] $message\u001B[0m")
+}
+
+def error(message) {
+    logger.error("\u001B[31m[PatchedArtifacts] $message\u001B[0m")
+}
+
+def warnIfNotConfigured(reason) {
+    warn("$reason")
+    warn("For setup instructions, refer to: https://github.com/Expensify/App?tab=readme-ov-file#enabling-prebuilt-react-native-artifacts-on-android")
+}
+
+def getNewDotRootDir() {
+    return "${rootDir}/${hasProperty('newDotRoot') ? getProperty('newDotRoot') : "/.."}"
+}
+
+def runCommand(command) {
+    def process = command.execute()
+    process.waitFor(10, TimeUnit.SECONDS)
+    return [output: process.in.text?.trim(), exitCode: process.exitValue()]
+}
+
+def hasGithubCLI() {
+    return runCommand(["which", "gh"]).exitCode == 0
+}
+
+def hasRequiredScopes() {
+    def scopes = runCommand(["gh", "auth", "status"]).output
+    return ['read:packages', 'write:packages'].any { scope -> scopes.contains(scope) }
+}
+
+def getGithubUsername() {
+    return new JsonSlurper().parseText(runCommand(["gh", "api", "user"]).output).login
```

Comment:
NAB but you initialize `JsonSlurper` inline several times - maybe just init it once at the top level of the file / in a class

Reply:
Yeah, this is limitation of groovy DSL and how it handles global scope so it was intentional, unfortunately.